analyzers/src/SonarAnalyzer.CSharp/Rules/ImplementISerializableCorrectly.cs

```diff
@@ -28,28 +28,35 @@
 using Microsoft.CodeAnalysis.Diagnostics;
 using SonarAnalyzer.Common;
 using SonarAnalyzer.Helpers;
+using StyleCop.Analyzers.Lightup;
 
 namespace SonarAnalyzer.Rules.CSharp
 {
     [DiagnosticAnalyzer(LanguageNames.CSharp)]
     [Rule(DiagnosticId)]
     public class ImplementISerializableCorrectly : SonarDiagnosticAnalyzer
     {
-        internal const string DiagnosticId = "S3925";
-        internal const string MessageFormat = "Update this implementation of 'ISerializable' to conform to the " +
-            "recommended serialization pattern.";
+        private const string DiagnosticId = "S3925";
+        private const string MessageFormat = "Update this implementation of 'ISerializable' to conform to the " +
+                                             "recommended serialization pattern.";
 
-        private static readonly DiagnosticDescriptor rule =
-            DiagnosticDescriptorBuilder.GetDescriptor(DiagnosticId, MessageFormat, RspecStrings.ResourceManager);
-        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(rule);
+        private static readonly DiagnosticDescriptor Rule = DiagnosticDescriptorBuilder.GetDescriptor(DiagnosticId, MessageFormat, RspecStrings.ResourceManager);
 
-        protected override void Initialize(SonarAnalysisContext context)
-        {
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(Rule);
+
+        protected override void Initialize(SonarAnalysisContext context) =>
             context.RegisterSyntaxNodeActionInNonGenerated(
                 c =>
                 {
-                    var classDeclaration = (ClassDeclarationSyntax)c.Node;
-                    var typeSymbol = c.SemanticModel.GetDeclaredSymbol(classDeclaration);
+                    if (c.ContainingSymbol.Kind != SymbolKind.NamedType)
+                    {
+                        return;
+                    }
+
+                    var typeDeclarationSyntax = (TypeDeclarationSyntax)c.Node;
+                    var typeKeyword = typeDeclarationSyntax.Keyword;
+                    var typeIdentifier = typeDeclarationSyntax.Identifier;
+                    var typeSymbol = c.SemanticModel.GetDeclaredSymbol(typeDeclarationSyntax);
```

Comment:
c.ContainingSymbol should be the same as the symbol returned by GetDeclaredSymbol, so you can use that instead.

Reply:
Good to know!